homeassistant/components/rainmachine/__init__.py

```diff
@@ -197,60 +232,71 @@
             refresh,
             timedelta(seconds=config_entry.data[CONF_SCAN_INTERVAL]))
 
-    async def disable_program(service):
+    async def disable_program(call):
         """Disable a program."""
+        _check_service_permissions(hass, call)
```

Comment:
You should probably write a test, so you would have noticed that you're not awaiting this async function 😉 

Also, consider making it a decorator, so you can write `@_check_servic_permissions` in front of each service handle function

Reply:
Example: https://github.com/home-assistant/home-assistant/blob/dev/homeassistant/components/websocket_api/decorators.py#L34-L46